pkg/api/client/dynamic/supported.go

```diff
@@ -0,0 +1,50 @@
+package dynamic
+
+import "strings"
+
+const (
+	PrometheusRuleKind = "PrometheusRule"
+	ServiceMonitorKind = "ServiceMonitor"
+)
+
+// SupportedResource is an interface that returns information about Dynamic Resource that OLM supports.
+type SupportedResource interface {
+	IsNamespaced() bool
+}
+
+// supportedResource contains information about Dynamic Resource that OLM supports.
+type supportedResource struct {
+	name       string
+	namespaced bool
+}
+
+// IsNamespaced returns whether or not the supportedResource is a namespaced object.
+func (sr supportedResource) IsNamespaced() bool {
```

Comment:
just a small nit: I think in Go its more iditomatic to drop off the `Is` especially when the function is returning a boolean. Simply `Namespaced()` shows the intent and using it like `if sr.Namespaced()` its clear to understand what the function is intended to show.